enhancements/catalog-source-config.md

```diff
@@ -0,0 +1,340 @@
+---
+title: catalogsource-config
+authors:
+  - "@perdasilva"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-11-18
+last-updated: 2021-11-18
+status: provisional
+---
+
+# catalogsource-config
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+
+### Open Questions
+
+I've put down the options as I see them. I need some input from the community to figure out the right way to go.
+The first two would required breaking changes and therefore migration to a new schema. The [first option](#Union-types)
+is better assuming we don't want to give users the option to implement controllers for additional, and custom,
+`CatalogSource` types. Otherwise, the [seconds option](#CatalogSourceClass) would be better. If we have bigger changes on
+the horizon, we may want to punt on the "right" approach, in favour of the [third option](#ImageConfig). This would 
+mitigate the issue, but not necessarily in the best way.
+
+I'm personally in favor of options 1 or 2, even though they will require more work, I think they provide an improvement
+in user experience.
+
+## Summary
+
+Currently, there is no way to override the pod spec for `grpc` type `CatalogSource` pods.
+This is desirable, for example, in cases where the admin would like additional control of 
+where in the infrastructure the catalog source pod get deployed. In this PE we
+propose changes to the `CatalogSource` spec to enable this kind customization.
+
+
+## Motivation
+
+This EP targets issue [#2452](https://github.com/operator-framework/operator-lifecycle-manager/issues/2452).
+Users would like to be able to override the pod spec of the CatalogSource pod
+created for `grpc` `CatalogSource`s. Especifically, they would like to be able to specify
+pod tolerations and node selectors.
+
+### Goals
+
+ * Update the `CatalogSource` spec to allow users to affect the pod spec of `grpc` `CatalogSource` pods 
```

Comment:
IHMO here would be more the use case. When/why does a user want to do that?
What is the user here? Is the Operator Authors, the catalog/index admin? What is the role?

Reply:
I'll fix this in the next iteration. Great call out.